src/kcpServer/game/gameData/data/MaterialData.ts

```diff
@@ -19,6 +21,19 @@ class MaterialDataLoader extends Loader {
   getMaterialList(): MaterialData[] {
     return this.data || []
   }
+
+  getStackLimit(path: string): { [key: number]: number } {
+    const data = fs.readFileSync(path, "utf8")
```

Comment:
Instead of reading the data file each time `getStackLimit` is called, you can use `getMaterialList` to get the data that was already loaded into memory